File: implementacao java/src/Dominio/Eleicao.java
package Dominio;
import java.time.LocalDate;
import java.util.HashMap;
import java.util.Map;

public class Eleicao {
    
    private Map<Integer, Candidato> candidatos = new HashMap<>();
    private Map<Integer, Partido> partidos = new HashMap<>();

    private int qtdCandidatosEleitos;
    private int qtdVotosNominais;
    private int qtdVotosLegenda;
    private int qtdMasculino;
    private int qtdFeminino;

    public Candidato criaCandidato(int codigo_municipio, int codigo_cargo, int numero_candidato, String nome_urna, int numero_federacao, LocalDate data_nascimento, boolean resultado_turno, Genero genero, Partido partido, LocalDate dataAtual){
        
        Candidato candidato = new Candidato(codigo_municipio, codigo_cargo, numero_candidato, nome_urna, numero_federacao, data_nascimento, resultado_turno, genero, partido, dataAtual );
        candidatos.put(numero_candidato, candidato);
        partido.adicionarCandidato(candidato);
        return candidato;
    }

    public Partido criaPartido(int numero_partido, String sigla_partido){
        
        Partido partido = this.partidos.get(numero_partido);
        
        if(partido == null){
            
           partido = new Partido(numero_partido, sigla_partido);
           partidos.put(numero_partido, partido);

        }

        return partido;
    }

    //                                                                getters
    
    public Map<Integer, Candidato> getCandidatos(){
        return new HashMap<>(candidatos);
    }

    public Candidato getCandidato(int numero_candidato){
        return  candidatos.get(numero_candidato);   
    }

    public Map<Integer, Partido> getPartidos(){
        return new HashMap<>(partidos);
    }

    public Partido getPartido(int numeroPartido){
        return  partidos.get(numeroPartido);   
    }

    public int getQtdVotosLegenda() {
        return qtdVotosLegenda;
    }

    public int getQtdVotosNominais() {
        return qtdVotosNominais;
    }

    public int getQtdCandidatosEleitos() {
        return qtdCandidatosEleitos;
    }

    public int getQtdFeminino() {
        return qtdFeminino;
    }

    public int getQtdMasculino() {
        return qtdMasculino;
    }

    //                                                                      incrementadores

    public void incrementarQtdVotosNominais(int votos){
        this.qtdVotosNominais+=votos;
    }

    public void incrementarQtdLegenda(int votos){
        this.qtdVotosLegenda+=votos;
    }

    public void incrementaQtdCandidatosEleitos(){
        this.qtdCandidatosEleitos++;
    }


    public void incrementrarQtdMasculino(){
        qtdMasculino++;
    }

    public void incrementrarQtdFeminino(){
        qtdFeminino++;
    }


}

File: implementacao c++/Eleicao.cpp
#include "Eleicao.hpp"
#include "Candidato.hpp"
#include "Partido.hpp"
#include <map>



Eleicao::Eleicao(){
    this->qtdVereadoresEleitos = 0;
    this->qtdFeminino =0;
    this->qtdMasculino=0;
    this->qtdVotosLegenda =0;
    this->qtdVotosNominais=0;
}


Candidato* Eleicao::criaCandidato(int codigoMunicipio, int codigoCargo, int numeroCandidato, const std::string &nomeUrna, int numeroFederacao, SituacaoCandidato resultadoTurno, Genero &genero,  Partido *partido,  Data& dataNascimento, Data &dataAtual){

    auto it = this->candidatos.find(numeroCandidato);

    
    if (it != this->candidatos.end()) {

        return it->second;
    }


    Candidato *candidato = new Candidato(codigoMunicipio, codigoCargo,numeroCandidato, nomeUrna,  numeroFederacao, resultadoTurno, genero, partido, dataNascimento, dataAtual);

    this->candidatos.insert({numeroCandidato, candidato});
    partido->adicionarCandidato(candidato);

    return candidato;
}


Candidato* Eleicao::getCandidato(int numeroCandidato)const{
     return candidatos.at(numeroCandidato);
}

const std::map<int, Candidato*>& Eleicao::getCandidatos()const{

    return this->candidatos;
}


Partido* Eleicao::criaPartido(int numeroPartido, const std::string &siglaPartido) {
    
   
    auto it = this->partidos.find(numeroPartido);

    
    if (it != this->partidos.end()) {

        return it->second;
    }

    Partido *partido = new Partido(numeroPartido, siglaPartido);
    this->partidos.insert({numeroPartido, partido});
    
    return partido;
}


Partido* Eleicao::getPartido(int numeroPartido)const{

    return partidos.at(numeroPartido);
}


const std::map<int, Partido*>& Eleicao::getPartidos()const{

    return this->partidos;
}



void Eleicao::incrementrarQtdMasculino(){
    this->qtdMasculino++;
}

void Eleicao::incrementrarQtdFeminino(){
    this->qtdFeminino++;
}

void Eleicao::incrementarQtdNominal(int qtdVotos){
    this->qtdVotosNominais+=qtdVotos;
}

void Eleicao::incrementarQtdLegenda(int qtdVotos){
    this->qtdVotosLegenda+=qtdVotos;
}
void Eleicao::incrementarVereadoresEleitos(){
    this->qtdVereadoresEleitos++;
}


int Eleicao::getQtdVereadoresEleitos()const{

    return this->qtdVereadoresEleitos;
}

int Eleicao::getQtdMasculino()const{

    return this->qtdMasculino;
    
}

int Eleicao::getQtdFeminino()const{

    return this->qtdFeminino;
}

int Eleicao::getQtdVotosNominais()const{

    return this->qtdVotosNominais;
    
}

int Eleicao::getQtdVotosLegenda()const{

    return this->qtdVotosLegenda;
    
}


 Eleicao::~Eleicao(){


    for(auto &par: this->partidos){
         delete par.second;
    }

    for(auto &par : this->candidatos){
        delete par.second;
    }
 }